### ISSUE TITLE
Batch-normalization - axis confusion

### ISSUE THREAD

**Issue by MaverickMeerkat** [keras-team/keras#13463]
When doing BN you can specify the axis. I was quite confused to discover that the axis you specify is actually the axis which **you don't do the operation on.** This is opposed to how common NumPy operations works where the axis you specify is the axis which you want to do the operation on. 

Is there a reason for this? 

This is very confusing, especially in the case of Conv-Nets, where you usually see the channels (features) axis specified as the axis. This makes very little sense, and you can find many questions online asking why is it so - with very few answers that actually explain that the axis in this case is treated in reversed.
